source/NVDAObjects/behaviors.py

```diff
@@ -365,6 +366,100 @@ def event_gainFocus(self):
 	def event_loseFocus(self):
 		self.stopMonitoring()
 
+
+class TerminalWithoutTypedCharDetection(Terminal):

```

Comment:
The class doesn't need events for typed character detection (it uses the `keyboardHandler` module for that), but rather for password suppression.

Reply:
I suggest `TerminalWithTypedCharSupportProvidedViaEvents` because it relies on NVDA events (not because of win events). 

I have had a conversation with @leonardder today, we went through this PR carefully and agreed that `KeyboardHandlerBasedTypedCharSupport` would be a more accurate name. Because it inherits from terminal, we don't think that it needs to state that in it's name as well.